sklearn/discriminant_analysis.py

```diff
@@ -819,6 +819,13 @@ class QuadraticDiscriminantAnalysis(ClassifierMixin, BaseEstimator):
     [1]
     """
 
+    _parameter_constraints = {
+        "priors": ["array-like", None],
+        "reg_param": [Interval(Real, 0, 1, closed="both")],
+        "store_covariance": ["boolean"],
+        "tol": [Interval(Real, 0, None, closed="left")],
+    }
+
     def __init__(
         self, *, priors=None, reg_param=0.0, store_covariance=False, tol=1.0e-4
     ):
```

Comment:
Oh we don't follow our own API here: we validate `priors` in the `__init__`.
Since this is about parameter validation, do you mind fixing it.

The idea is to have

```python
def __init__(...):
    self.priors = priors

def fit(...):
    ...
    if self.priors is not None:
        priors = np.array(priors, copy=False)
    ...
```

You can also update the docstring of `priors` by changing `ndarray` to array-like.
It could be nice to check that we have a test in `test_discriminant_analysis.py` that is parametrize with a python list and a numpy array when passing `priors` explicitely.

Reply:
Done. I also included the what's new entry per your other comment.